plugins/catalog-backend/src/next/Stitcher.test.ts

```diff
@@ -26,181 +26,200 @@ import { DbSearchRow } from './search';
 import { DbFinalEntitiesRow, Stitcher } from './Stitcher';
 
 describe('Stitcher', () => {
-  let db: Knex;
+  const databases = TestDatabases.create();
   const logger = getVoidLogger();
 
-  beforeEach(async () => {
-    db = await DatabaseManager.createTestDatabaseConnection();
-    await DatabaseManager.createDatabase(db);
-  });
-
-  it('runs the happy path', async () => {
-    const stitcher = new Stitcher(db, logger);
-
-    await db.transaction(async tx => {
-      await tx<DbRefreshStateRow>('refresh_state').insert([
-        {
-          entity_id: 'my-id',
-          entity_ref: 'k:ns/n',
-          unprocessed_entity: JSON.stringify({}),
-          processed_entity: JSON.stringify({
-            apiVersion: 'a',
-            kind: 'k',
-            metadata: {
-              name: 'n',
-              namespace: 'ns',
-            },
-            spec: {
-              k: 'v',
-            },
-          }),
-          errors: '[]',
-          next_update_at: tx.fn.now(),
-          last_discovery_at: tx.fn.now(),
-        },
-      ]);
-      await tx<DbRefreshStateReferencesRow>('refresh_state_references').insert([
-        { source_key: 'a', target_entity_ref: 'k:ns/n' },
-      ]);
-      await tx<DbRelationsRow>('relations').insert([
-        {
-          originating_entity_id: 'my-id',
-          source_entity_ref: 'k:ns/n',
-          type: 'looksAt',
-          target_entity_ref: 'k:ns/other',
-        },
-      ]);
-    });
-
-    await stitcher.stitch(new Set(['k:ns/n']));
-
-    let firstHash: string;
-    await db.transaction(async tx => {
-      const entities = await tx<DbFinalEntitiesRow>('final_entities');
-
-      expect(entities.length).toBe(1);
-      const entity = JSON.parse(entities[0].final_entity);
-      expect(entity).toEqual({
-        relations: [
+  it.each([
+    ['POSTGRES_13'],
+    ['POSTGRES_9'],
+    ['SQLITE_3'],
+    // TODO(freben): mysql:8 is not ready to use yet
+  ] as const)(
```

Comment:
This bit is where we talked about making it callback-based instead right?

Reply:
yass exactly